src/status_im2/contexts/chat/composer/link_preview/events_test.cljs

```diff
@@ -0,0 +1,212 @@
+(ns status-im2.contexts.chat.composer.link-preview.events-test
+  (:require [status-im2.contexts.chat.composer.link-preview.events :as events]
+            [cljs.test :refer [is deftest testing]]))
+
+(def url-github "https://github.com")
+(def url-gitlab "https://gitlab.com")
+(def preview-github {:url url-github :thumbnail nil})
+(def preview-gitlab {:url url-gitlab :thumbnail nil})
+(def request-id "fb973113-86f3-49c0-b750-5734a2e77a90")
+
+(deftest unfurl-urls
+  (testing "clear up state when text is empty"
+    (let [cofx                       {:db {:chat/link-previews {:unfurled   {}
+                                                                :cache      {}
+                                                                :request-id "123"}}}
+          {:keys [db json-rpc/call]} (events/unfurl-urls cofx " ")]
+      (is (= {:chat/link-previews {:cache {}}}
+             db))
+      (is (nil? call))))
+
+  (testing "fetches parsed URLs"
+    (let [cofx                       {:db {:chat/link-previews {:unfurled   {}
+                                                                :cache      {}
+                                                                :request-id "123"}}}
+          {:keys [db json-rpc/call]} (events/unfurl-urls cofx url-github)]
+      (is (nil? db))
+      (is (= {:params [url-github]}
+             (select-keys (first call) [:params]))))))
+
+(defn assert-request-id
+  [db]
+  (is (string? (get-in db [:chat/link-previews :request-id]))))
```

Comment:
If it were me and I had time, I would convert that `request-id (str (random-uuid))` line to its own `defn generate random-uuid` fn, and then use `with-redefs` here in order to insert a test uuid like "123". The tests here could just use assert maps with "123" instead of using this `assert-request-id` fn.

Reply:
That's a great idea!